com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -11,364 +11,217 @@ namespace MLAPI.Prototyping
-                    }
-                }
-            }
-
-            if (IsServer && EnableRange && EnableNonProvokedResendChecks)
+            m_PositionChangedDelegate = GetOnValueChangedDelegate<Vector3>(current =>
             {
-                CheckForMissedSends();
-            }
-        }
+                transform.position = current;
+                m_OldPosition = current;
+            });
+            m_NetworkPosition.OnValueChanged += m_PositionChangedDelegate;
+            m_RotationChangedDelegate = GetOnValueChangedDelegate<Quaternion>(current =>
+            {
+                transform.rotation = current;
+                m_OldRotation = current;
+            });
+            m_NetworkRotation.OnValueChanged += m_RotationChangedDelegate;
+            m_ScaleChangedDelegate = GetOnValueChangedDelegate<Vector3>(current =>
+            {
+                SetWorldScale(current);
+                m_OldScale = current;
+            });
+            m_NetworkWorldScale.OnValueChanged += m_ScaleChangedDelegate;
 
-        [ClientRpc]
-        private void ApplyTransformClientRpc(Vector3 position, Vector3 eulerAngles, ClientRpcParams rpcParams = default)
-        {
-            if (enabled)
+            if (IsServer)
             {
-                ApplyTransformInternal(position, Quaternion.Euler(eulerAngles));
+                if (authority == Authority.Client)
+                {
+                    m_NetworkPosition.Settings.WritePermission = NetworkVariablePermission.OwnerOnly;
+                    m_NetworkRotation.Settings.WritePermission = NetworkVariablePermission.OwnerOnly;
+                    m_NetworkWorldScale.Settings.WritePermission = NetworkVariablePermission.OwnerOnly;
+                }
+                else if (authority == Authority.Shared)
+                {
+                    m_NetworkPosition.Settings.WritePermission = NetworkVariablePermission.Everyone;
+                    m_NetworkRotation.Settings.WritePermission = NetworkVariablePermission.Everyone;
+                    m_NetworkWorldScale.Settings.WritePermission = NetworkVariablePermission.Everyone;
+                }
             }
         }
 
-        private void ApplyTransformInternal(Vector3 position, Quaternion rotation)
+        public void OnDestroy()
         {
-            if (!enabled)
-            {
-                return;
-            }
-
-            if (InterpolatePosition && (!IsServer || InterpolateServer))
-            {
-                m_LastReceiveTime = Time.unscaledTime;
-                m_LerpStartPos = transform.position;
-                m_LerpStartRot = transform.rotation;
-                m_LerpEndPos = position;
-                m_LerpEndRot = rotation;
-                m_LerpTime = 0;
-            }
-            else
-            {
-                transform.position = position;
-                transform.rotation = rotation;
-            }
+            m_NetworkPosition.OnValueChanged -= m_PositionChangedDelegate;
```

Comment:
This is problematic. We subscribe in `NetworkStart` but unsubscribe in `Destroy`. If we spawn then unspawn then spawn an object again NetworkStart gets called multiple times which means the event will fire multiple times.

Reply:
Good catch, will put registration in start instead of network start.

Reply:
The registration is still in networkstart?